### ISSUE TITLE
Fluent Bit crashes when parsing invalid JSON

### ISSUE THREAD

**Issue by rolandjitsu** [fluent/fluent-bit#1489]
## Bug Report

**Describe the bug**

Tailing a file that has invalid JSON will make Fluent Bit crash.

**To Reproduce**
- Rubular link if applicable:
- Example log message if applicable:
```
{ "datetime":"2019-05-31T07:19:17.087" , "event":"some event", "msg"="bad formatted json" }
{ "datetime":"2019-05-31T07:19:17.092" , "event":"another event"}
Some odd log message that is not JSON ...
{ "datetime":"2019-05-31T07:19:22.200" "event":"and another one", "msg":"another badly formatted json line" }

100% [||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| ]
{ "datetime":"2019-05-31T07:19:23.496" , "event":"some event", "percentage":6, "eta":20 }
 97% [||||||||||||||||||||||||||||||||||||||||||||||||||||||||||  ]
{ "datetime":"2019-05-31T07:19:23.496" , "event":"some event", "percentage":7, "eta":19 }100% [||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| ]
{ "datetime":"2019-05-31T07:19:23.496" , "event":"some event", "percentage":9, "eta":17 }
```
- Steps to reproduce the problem:
  * Run fluent bit with
```
fluent-bit -c fluent-bit.conf
```

**Expected behavior**

It should just skip badly formatted lines instead of crashing.

**Screenshots**

**Your Environment**

* Version used: 1.2.0
* Configuration:
```
# fluent-bit.conf
[SERVICE]
    Flush              5
    HTTP_Server Off
    Daemon         Off
    Log_Level      info
    Log_File         /home/pi/logs/fluent-bit.log
    Parsers_File   parsers.conf
    Streams_File    streams.conf

    # https://docs.fluentbit.io/manual/configuration/buffering
    storage.path                           /home/pi/fb-buf/
    storage.sync                          normal
    storage.checksum                 on
    storage.backlog.mem_limit   10M

@INCLUDE inputs.conf
@INCLUDE filters.conf
@INCLUDE outputs.conf

# inputs.conf
[INPUT]
    Name                   tail
    Alias                     test_src
    Tag                       test_src
    Path                     /home/pi/logs/test.log
    Parser                  el_json
    DB                        /home/pi/fbit.db
    Mem_Buf_Limit   5M
    Routable        false

# filters.conf
[FILTER]
    Name    modify
    Alias      add_hostname
    Match   *
    Condition Key_does_not_exist hostname
    Add  hostname ${HOSTNAME}

# outputs.conf
[OUTPUT]
    Name    stdout
    Alias      stdout
    Match   *

# parsers.conf
[PARSER]
    Name          el_json
    Format        json
    Time_Key     datetime
    Time_Format  %Y-%m-%dT%H:%M:%S.%L
    Time_Keep    Off

# streams.conf
# CPU
[STREAM_TASK]
    Name   test_stream
    Exec   CREATE STREAM test_stream WITH (tag='test') AS SELECT datetime AS d, event AS e, msg AS m FROM TAG:'test_src';
```
* Environment name and version (e.g. Kubernetes? What version?):
* Server type and version:
* Operating System and version:
```
Linux testpi 4.14.80-v7+ #1161 SMP Mon Nov 12 18:51:43 GMT 2018 armv7l GNU/Linux

PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```
* Filters and plugins:

**Additional context**

I understand why FB crashes, but I think it should just log the error and continue with the next line instead of crashing. It's very difficult to tell why it crashed if you have many files that are tailed so it would be helpful if it only logs the line which could not be parsed as JSON and move on.

**Comment by nigels-com** [comment]
It exits with an error, or simply crashes without explanation?

**Comment by rolandjitsu** [comment]
Just exists, no error.

**Comment by nigels-com** [comment]
I had a go at reproducing this, but no luck.

I'd suggest putting together a self-contained set of files (in one directory) that doesn't require any guesswork.

Simplifying as much as possible also helps.

Thanks.

```
$ valgrind -- build/bin/fluent-bit -c config.cfg 
==26126== Memcheck, a memory error detector
==26126== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==26126== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==26126== Command: build/bin/fluent-bit -c config.cfg
==26126== 
Fluent Bit v1.2.0
Copyright (C) Treasure Data

[2019/08/02 23:30:56] [debug] [storage] [cio stream] new stream registered: test_src
[2019/08/02 23:30:56] [ info] [storage] initializing...
[2019/08/02 23:30:56] [ info] [storage] in-memory
[2019/08/02 23:30:56] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2019/08/02 23:30:56] [ info] [engine] started (pid=26126)
[2019/08/02 23:30:56] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2019/08/02 23:30:56] [debug] [in_tail] inotify watch fd=19
[2019/08/02 23:30:56] [debug] [in_tail] scanning path test.log
[2019/08/02 23:30:56] [debug] [in_tail] add to scan queue test.log, offset=0
[2019/08/02 23:30:56] [debug] [filter_modify] Initialized modify filter with 1 conditions and 1 rules
[2019/08/02 23:30:56] [debug] [router] match rule tail.0:stdout.0
[2019/08/02 23:30:56] [debug] [storage] [cio stream] new stream registered: test_stream
[2019/08/02 23:30:56] [ info] [sp] stream processor started
[2019/08/02 23:30:56] [ info] [sp] registered task: test_stream
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 3 elements, output map size 4 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 3 elements, output map size 4 elements
[2019/08/02 23:30:56] [debug] [in_tail] file=test.log read=786 lines=9
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [filter_modify] Input map size 1 elements, output map size 2 elements
[2019/08/02 23:30:56] [debug] [in_tail] file=test.log promote to TAIL_EVENT
[0] test_src: [1564752656.550544008, {"log"=>"{ "datetime":"2019-05-31T07:19:17.087" , "event":"some event", "msg"="bad formatted json" }", "hostname"=>"zooropa"}]
[1] test_src: [1559287157.092000000, {"event"=>"another event", "hostname"=>"zooropa"}]
[2] test_src: [1564752656.585243784, {"log"=>"Some odd log message that is not JSON ...", "hostname"=>"zooropa"}]
[3] test_src: [1564752656.585362321, {"log"=>"{ "datetime":"2019-05-31T07:19:22.200" "event":"and another one", "msg":"another badly formatted json line" }", "hostname"=>"zooropa"}]
[2019/08/02 23:31:01] [debug] [task] created task=0x5f6c4e0 id=0 OK
[4] test_src: [1564752656.587161457, {"log"=>"100% [||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| ]", "hostname"=>"zooropa"}]
[5] test_src: [1559287163.496000000, {"event"=>"some event", "percentage"=>6, "eta"=>20, "hostname"=>"zooropa"}]
[6] test_src: [1564752656.589624416, {"log"=>" 97% [||||||||||||||||||||||||||||||||||||||||||||||||||||||||||  ]", "hostname"=>"zooropa"}]
[7] test_src: [1564752656.589677271, {"log"=>"{ "datetime":"2019-05-31T07:19:23.496" , "event":"some event", "percentage":7, "eta":19 }100% [||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| ]", "hostname"=>"zooropa"}]
[8] test_src: [1559287163.496000000, {"event"=>"some event", "percentage"=>9, "eta"=>17, "hostname"=>"zooropa"}]
[0] test: [1559287157.092000000, {"e"=>"another event", "hostname"=>"zooropa"}]
[1] test: [1559287163.496000000, {"e"=>"some event", "hostname"=>"zooropa"}]
[2019/08/02 23:31:01] [debug] [task] created task=0x5f85550 id=1 OK
[2] test: [1559287163.496000000, {"e"=>"some event", "hostname"=>"zooropa"}]
[2019/08/02 23:31:01] [debug] [task] destroy task=0x5f6c4e0 (task_id=0)
[2019/08/02 23:31:01] [debug] [task] destroy task=0x5f85550 (task_id=1)
^C[engine] caught signal (SIGINT)
[2019/08/02 23:31:06] [ info] [input] pausing tail.0
[2019/08/02 23:31:06] [ info] [input] pausing stream_processor.1
==26126== 
==26126== HEAP SUMMARY:
==26126==     in use at exit: 0 bytes in 0 blocks
==26126==   total heap usage: 580 allocs, 580 frees, 1,461,864 bytes allocated
==26126== 
==26126== All heap blocks were freed -- no leaks are possible
==26126== 
==26126== For counts of detected and suppressed errors, rerun with: -v
==26126== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

```
$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.2 LTS"

$ uname -a
Linux zooropa 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by rolandjitsu** [comment]
@nigels-com I will try that. Though, I haven't seen this happening with the latest version (v1.2.2).

**Comment by nigels-com** [comment]
Presumably fixed since 1.2.0?
